### PR TITLE
Codestyle unification

### PR DESCRIPTION
**Pull request by sladg**
What this does?

- Pnpm-lock version changed to 6.0, see: https://github.com/pnpm/pnpm/issues/6648, lock files updated,
- Removed yarn references, pnpm to be used across repo,
- Eslint and prettier rules added, so we can lint files on PRs and ensure codebase stays formatted,
- Removed .sst folder which is auto-generated and included personal data.
- Fixes https://github.com/sst/open-next/issues/174

### CODE REVIEW

**Review by khuezy — APPROVED** [review]
Thanks! Everything is style/cosmetic except the image opt stream right? I'll test this in a bit then we can merge.

File: packages/open-next/src/adapters/image-optimization-adapter.ts [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
   imageOptimizer,
   ImageOptimizerCache,
   // @ts-ignore
 } from "next/dist/server/image-optimizer";
+// @ts-ignore
+import type { NextUrlWithParsedQuery } from "next/dist/server/request-meta";
+
+import { RequestHandler } from "../types.js";
+import { awsLogger, debug, error } from "./logger.js";
+import { streamError, streamSuccess } from "./streaming.js";
```
Comment (khuezy) [inline packages/open-next/src/adapters/image-optimization-adapter.ts:26]: @sladg I'm getting type errors when I run `npm run build` in the `packages/open-next` folder. I don't see the `types.ts` or `streaming.ts` files in this PR. Did you mean to include your streaming changes in this PR? If so, you might have missed the files.

**Review by khuezy — CHANGES_REQUESTED** [review]
I think there's a couple of uncommitted files

File: packages/open-next/src/adapters/image-optimization-adapter.ts (hunk 1 above)
Comment (sladg) [reply packages/open-next/src/adapters/image-optimization-adapter.ts:26]: Thanks for heads up! Seems like bad rebase from `main` branch.

**Comment by khuezy** [comment]
@sladg is there anything special I need to do to get the autoformat to work? I have eslint and prettier plugin installed in my VSCode, but when I save a file, it doesn't add the commas.
The eslint is picking up the rules but it's not applying the changes on save.

**Comment by sladg** [comment]
@khuezy disable prettier plugin. Only eslint is needed as prettier does not have configuration file in workspace so it's fighting with eslint most likely.

**Comment by khuezy** [comment]
I think I have some weird VSCode settings in my global VSCode settings somewhere in my /Library/Application Support/Code/User. It might be better to create a `.vscode/settings.json` file to force the project to adhere to open-next's VSCode settings.

**Comment by khuezy** [comment]
My VSCode broke... so I can't really test this PR smh...

**Comment by khuezy** [comment]
I was using typescript 5.2.2 and the eslint lib didn't support it yet. The fix got deployed ~30m ago and everything is working after I reinstall to pick up new packages. See: https://github.com/typescript-eslint/typescript-eslint/issues/7155#issuecomment-1693975417

**Comment by khuezy** [comment]
Do you mind adding `.vscode/settings.json`:
```
{
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
  },
}
```
That will autoformat to the style on save.

**Comment by khuezy** [comment]
I also had to make some changes to server-adapter.ts. Let me know if you want me to push the changes up or not.

**Review by khuezy — APPROVED** [review]
LGTM Thanks!

**Pull request merged**